src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.AsyncWorkItemQueue.cs

```diff
@@ -259,26 +258,17 @@ protected CancellationToken GetNewCancellationToken_NoLock(object key)
                             return projectId;
                         }
 
-                        // prefer project that directly depends on the given project and has diagnostics as next project to
+                        // prefer project that directly depends on the given project as next project to
                         // process
                         foreach (var dependingProjectId in dependencyGraph.GetProjectsThatDirectlyDependOnThisProject(projectId))
                         {
-                            if (workQueue.ContainsKey(dependingProjectId) && analyzerService?.ContainsDiagnostics(Workspace, dependingProjectId) == true)
+                            if (workQueue.ContainsKey(dependingProjectId))
                             {
                                 return dependingProjectId;
                             }
                         }
                     }
 
-                    // prefer a project that has diagnostics as next project to process.
```

Comment:
could you file an issue saying that we should consider walkign in this order in workspace pull diags.  We have the information about the previous result, so we should be able to add this back in.  Personally, i do think it's a sensible heuristic.  But i am totally ok with it not being deep in the engine, but a policy decision of the system that 'pulls'.

Reply:
#67687